buildSrc/src/main/java/org/opensearch/gradle/docker/DockerSupportService.java

```diff
@@ -123,15 +122,11 @@ public DockerAvailability getDockerAvailability() {
 
                     if (isVersionHighEnough) {
                         // Check that we can execute a privileged command
-                        lastResult = runCommand(dockerPath, "images");
-
+                        lastResult = runCommand(execOperations, dockerPath, "images");
                         // If docker all checks out, see if docker-compose is available and working
-                        Optional<String> composePath = getDockerComposePath();
-                        if (lastResult.isSuccess() && composePath.isPresent()) {
-                            isComposeAvailable = runCommand(composePath.get(), "version").isSuccess();
+                        if (lastResult.isSuccess()) {
+                            dockerComposeAvailability = DockerComposeAvailability.detect(execOperations, dockerPath).orElse(null);
```

Comment:
@cwperks slightly changing the logic to be predictable: if there are two docker compose versions available, v1 and v2, always picking v1, otherwise we don't have any other mechanism to alter the selection. If user wants v2, the v1 could be easily uninstalled (but not v2 since it is part of the docker command line / plugin).